root/static/scripts/relationship-editor/components/DialogEntityCredit.js

```diff
@@ -190,35 +204,35 @@ const DialogEntityCredit = (React.memo<PropsT, void>(({
     );
   }
 
+  const helpText = l(
+    `A credited name is optional. You can leave this field blank
+     to keep the current name.`,
+  );
+
   return (
-    <tr className={forEntity + '-entity-credit'}>
-      <td className="section">
-        <label
-          className="credit-field"
-          htmlFor={inputId}
-        >
-          {addColonText('Credited as')}
-        </label>
-      </td>
-      <td className="fields">
-        <input
-          className="entity-credit"
-          id={inputId}
-          onChange={handleCreditedAsChange}
-          placeholder={entityName}
-          ref={inputRef}
-          type="text"
-          value={state.creditedAs}
-        />
-        <HelpIcon
-          content={l(
-            `A credited name is optional. You can leave this field blank
-             to keep the current name.`,
-          )}
-        />
-        {changeCreditsSection}
-      </td>
-    </tr>
+    <div className={forEntity + '-entity-credit'}>
+      {isInputVisible ? (
+        <>
+          <input
+            aria-description={helpText}
+            aria-label={l('Credited as')}
+            className="entity-credit"
+            id={inputId}
+            onChange={handleCreditedAsChange}
+            placeholder={entityName}
+            ref={inputRef}
+            type="text"
+            value={state.creditedAs}
+          />
+          <HelpIcon content={helpText} />
+          {changeCreditsSection}
+        </>
```

Comment:
Hmm. If we want to hide "Credited as" behind a click, I'd say it should be called "Add credit" or something, not "Credited as", which is not an action.

That said, since the box is already more compact now once you click, wouldn't it be easier to have them always on, but taking less space as they do now? I don't feel the button is saving any space now and it asks for an extra click. Something like this:
![Screenshot from 2022-09-27 12-31-54](https://user-images.githubusercontent.com/1069224/192490452-e7ce057b-3dc3-48b7-a0c0-a3a3e58d26b3.png)

```suggestion
        <>
          <label
            className="credit-field"
            htmlFor={inputId}
          >
            {addColonText('Credited as')}
          </label>
          {' '}
          <input
            aria-description={helpText}
            aria-label={l('Credited as')}
            className="entity-credit"
            id={inputId}
            onChange={handleCreditedAsChange}
            placeholder={entityName}
            ref={inputRef}
            type="text"
            value={state.creditedAs}
          />
          <HelpIcon content={helpText} />
          {changeCreditsSection}
        </>
```

See also there's another bug here, added MBS-12640 for it :)

Reply:
Thanks for trying it!

I was mainly trying to reduce visual clutter with the button.  I'm fine with keeping the input visible, but then not sure how MBS-12635 is resolved.  It's true that I did remove some padding/border-spacing, but that doesn't seem to be what the ticket is asking.  (More like remove the input entirely, so I tried to find a middle ground.)

Clicking the button does auto-focus the input at least, but if you use the keyboard it does require an extra keystroke.

The only thing I don't like about the old (production) layout is that the size and position of the credited-as input depends on the length of the label text ("Credited as:"), which also varies per language.  So it looks a bit chaotic to me, since it's not aligned or sized consistently.  That's why I moved the "Credited as:" label to go in the left column before.

What if we also put the relationship credit inputs to the right of the target autocomplete/source entity link, just like the instruments have it?

Reply:
That might work. I have no idea why that dialog is so small anyway - it's already too big to fit on a phone comfortably anyway, so why not make it take the space it needs :)

Reply:
I'm also fine with not fully solving MBS-12635 anyway - I think the main issue was just how prominent these were, since the credit fields were already shown before, right?

Reply:
Moved the credit inputs back to under the fields, though I dropped the label and set the placeholder to "Credited as" instead, like the instrument credits have it.